src/lambda.c

```diff
@@ -30,52 +31,6 @@
 #define ROUND(x)    (floor((x)+0.5))
 #define SWAP(x,y)   do {double tmp_; tmp_=x; x=y; y=tmp_;} while (0)
 
-
-static void triu2(u32 n, double *M)
-{
-  for (u32 i=1; i<n; i++) {
-    for (u32 j=0; j<i; j++) {
-      M[i*n + j] = 0;
-    }
-  }
-}
-
-static void eye2(u32 n, double *M)
-{
-  memset(M, 0, n * n * sizeof(double));
-  for (u32 i=0; i<n; i++) {
-    M[i*n + i] = 1;
-  }
-}
-
-static s8 udu2(u32 n, double *M, double *U, double *D) //todo: replace with DSYTRF
-{
-  double alpha, beta;
-  triu2(n, M);
-  eye2(n, U);
-  memset(D, 0, n * sizeof(double));
-
-  for (u32 j=n; j>=2; j--) {
-    D[j - 1] = M[(j-1)*n + j-1];
-    if (D[j-1] > 0) {
```

Comment:
The matrix_udu function to replace this has "D[j-1] != 0" instead of "D[j-1] > 0" here. Before I sign off, I want to check which one stays positive semidefinite when we get really close to singular.